tests/unit/src/test/scala/tests/codeactions/FlatMapToForComprehensionSuite.scala

```diff
@@ -0,0 +1,43 @@
+package tests.codeactions
+
+import scala.meta.internal.metals.codeactions.FlatMapToForComprehensionCodeAction
+import scala.meta.internal.metals.codeactions.RewriteBracesParensCodeAction
+
+class FlatMapToForComprehensionSuite
+    extends BaseCodeActionLspSuite("forComprehension") {
+
+  check(
+    "two-flatMaps-to-forComprehension",
+    """|object A {
+       |  val a: List[Int] = List(1, 2)
+       |
+       |  val b: Option[String] = Some("great")
+       |
+       |  val c = a.fl<<>>atMap { aMember =>
+       |                        b.map( bMember =>
+       |                            s"$aMember $bMember"
+       |                              )
+       |                     }
+       |
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toParens}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |""".stripMargin,
+    """|object A {
```

Comment:
Let's try and add some more complex scenarios, you can also create a side test that mocks Trees etc.

Reply:
Definitely!